eis_toolkit/raster_processing/reclassify_raster.py

```diff
@@ -0,0 +1,484 @@
+                    & (data_array != np.nan)
+                )
+            ] = (j + 1)
+            values_out[
+                np.where(
+                    ((median_value - width[j]) > data_array)
+                    & (data_array >= (median_value - width[j + 1]))
+                    & (data_array != np.nan)
+                )
+            ] = (-j - 1)
+            k = j
+
+        values_out[np.where(((median_value + width[k + 1]) < data_array) & (data_array != np.nan))] = k + 1
+        values_out[np.where(((median_value - width[k + 1]) > data_array) & (data_array != np.nan))] = -k - 1
+        values_out[np.where(median_value == data_array)] = 0
+
+        out_image.append(values_out)
+
+    return out_image, out_meta
+
+
+@beartype
+def raster_with_geometrical_intervals(  # type: ignore[no-any-unimported]
+    raster: rasterio.io.DatasetReader, number_of_classes: int, bands: Optional[Sequence[int]] = None
+) -> Tuple[Sequence[np.ndarray], dict]:
+    """Classify raster with geometrical intervals (Torppa, 2023).
+
+    If bands are not given, all bands are used for classification.
+
+    Args:
+        raster: Raster to be classified.
+        number_of_classes: The number of classes. The true number of classes is at most double the amount,
+        depending how symmetrical the input data is.
+        bands: Selected bands from multiband raster. Indexing begins from one. Defaults to None.
+
+    Returns:
+        Raster classified with geometrical intervals (Torppa, 2023) and metadata.
+    """
+    if bands is not None:
+        if not isinstance(bands, list):
+            raise InvalidParameterValueException("Expected bands parameter to be a list")
+        elif not all(isinstance(band, int) for band in bands):
+            raise InvalidParameterValueException("Expected bands to be a list of integers")
+        elif len(bands) > raster.count:
+            raise InvalidParameterValueException("The number of bands given exceeds the number of raster's bands")
+    if number_of_classes == 0:
+        raise InvalidParameterValueException("number_of_classes parameter is 0")
+
+    out_image, out_meta = _raster_with_geometrical_intervals(raster, number_of_classes, bands)
+
+    return out_image, out_meta
+
+
+def _raster_with_standard_deviation(  # type: ignore[no-any-unimported]
+    raster: rasterio.io.DatasetReader,
+    number_of_intervals: int,
+    bands: Optional[Sequence[int]] = None,
+) -> Tuple[Sequence[np.ndarray], dict]:
+
+    out_image: Sequence = []
+    out_meta = raster.meta.copy()
+
+    band_statistics = []
+    if bands is not None:
+        for band in bands:
+            stddev = np.nanstd(band)
+            mean = np.nanmean(band)
+            band_statistics.append((mean, stddev))
+    else:
+        for band in range(1, raster.count + 1):
+            stddev = np.nanstd(band)
+            mean = np.nanmean(band)
+            band_statistics.append((mean, stddev))
```

Comment:
The type of variable band is integer so you are taking np.nanstd and np.nanmean of an integer which probably was not the intention?